src/hammer-vlsi/hammer_driver.py

```diff
@@ -148,6 +153,80 @@ def update_tool_configs(self) -> None:
         tools = reduce(lambda a, b: a + b, list(self.tool_configs.values()))
         self.database.update_tools(tools)
 
+    def load_formal_tool(self, run_dir: str = "") -> bool:
+        """
+        Load the place and route tool based on the given database.
+
+        :param run_dir: Directory to use for the tool run_dir. Defaults to the run_dir passed in the HammerDriver
+                        constructor.
+        """
+        if run_dir == "":
+            # Use the synthesis directory for formal tools
+            run_dir = os.path.join(self.obj_dir, "syn-rundir")
+
+        formal_tool_name = self.database.get_setting("vlsi.core.formal_tool")
+        formal_tool_get = load_tool(
+            path=self.database.get_setting("vlsi.core.formal_tool_path"),
+            tool_name=formal_tool_name
+        )
+        assert isinstance(formal_tool_get, HammerFormalTool), "Formal tool must be a HammerFormalTool"
+        formal_tool = formal_tool_get  # type: HammerTool
+        formal_tool.name = formal_tool_name
+        formal_tool.logger = self.log.context("formal")
+        formal_tool.set_database(self.database)
+        formal_tool.run_dir = run_dir
+
+        formal_tool.input_files = self.database.get_setting("synthesis.inputs.input_files")
```

Comment:
I don't think this overloading makes sense? But I'm not sure we need another option. Shouldn't the reference be the synthesis input and the implementation be the synthesis output?